yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/YamlTestExtension.java

```diff
@@ -0,0 +1,161 @@
+            try {
+                config.afterAll();
+                return null;
+            } catch (Exception e) {
+                return e;
+            }
+        }).filter(Objects::nonNull).findFirst();
+        if (exception.isPresent()) {
+            throw exception.get();
+        }
+    }
+
+    @Override
+    public boolean supportsTestTemplate(final ExtensionContext context) {
+        return true; // TODO check type
+    }
+
+    @Override
+    public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(final ExtensionContext context) {
+        return testConfigs.stream().map(config -> new Context(config, context.getRequiredTestMethod().getAnnotation(ExcludeYamlTestConfig.class)));
+    }
+
+    /**
+     * Context for an individual test run (method + config).
+     */
+    private static class Context implements TestTemplateInvocationContext {
+        private final YamlTestConfig config;
+        private final Set<Class<? extends YamlTestConfig>> excludedConfigs;
+
+        public Context(final YamlTestConfig config, final ExcludeYamlTestConfig excludedConfigs) {
+            this.config = config;
+            this.excludedConfigs = excludedConfigs == null ? Set.of() : Set.of(excludedConfigs.value());
+        }
+
+
+        @Override
+        public String getDisplayName(int invocationIndex) {
+            return config.toString();
+        }
+
+        @Override
+        public List<Extension> getAdditionalExtensions() {
+            return List.of(new ClassParameterResolver(config, excludedConfigs));
+        }
+    }
+
+    /**
+     * Parameter resolver for the class when injecting the {@link YamlTest.Runner}.
+     */
+    private static final class ClassParameterResolver implements ParameterResolver {
+
+        private final YamlTestConfig config;
+        private final Set<Class<? extends YamlTestConfig>> excludedConfigs;
+
+        public ClassParameterResolver(YamlTestConfig config, final Set<Class<? extends YamlTestConfig>> excludedConfigs) {
+            this.config = config;
+            this.excludedConfigs = excludedConfigs;
+        }
+
+        @Override
+        public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+            return parameterContext.getParameter().getType().equals(YamlTest.Runner.class);
+        }
+
+        @Override
+        public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+            return new YamlTest.Runner() {
+                @Override
+                public void run(final String fileName) throws Exception {
+                    run(fileName, false);
+                }
+
+                @Override
+                public void run(final String fileName, final boolean correctExplain) throws Exception {
+                    config.assumeSupport(fileName, excludedConfigs);
```

Comment:
Why do we need this? I'm concerned that tests would get skipped but there won't be an obvious reason for it from the test code.